### PR TITLE
compile/opt: add config DEBUG_OPT_UNUSED_SECTIONS

### PR DESCRIPTION
**Pull request by anchao**
## Summary

compile/opt: add config DEBUG_OPT_UNUSED_SECTIONS

Enable this option to optimization the unused input sections with the
linker by compiling with " -ffunction-sections -fdata-sections ", and
linking with " --gc-sections ".

Signed-off-by: chao.an <anchao@xiaomi.com>

## Impact

Compile optimization

## Testing

CI-Check

### CODE REVIEW

**Review by pkarashchenko — APPROVED** [review]

**Comment by pkarashchenko** [comment]
Seems that something is wrong on mainline
```
riscv64-unknown-elf-ld: /Users/runner/work/incubator-nuttx/incubator-nuttx/sources/nuttx/staging/libarch.a(bl602_hbn.o): in function `bl602_hbn_irq':
/Users/runner/work/incubator-nuttx/incubator-nuttx/sources/nuttx/arch/risc-v/src/chip/bl602_hbn.c:91: undefined reference to `bl602_hbn_clear_rtc_int'
make[1]: *** [nuttx] Error 1
make: *** [nuttx] Error 2
make: Target `all' not remade because of errors.
```

**Comment by anchao** [comment]
> ```
> riscv64-unknown-elf-ld: /Users/runner/work/incubator-nuttx/incubator-nuttx/sources/nuttx/staging/libarch.a(bl602_hbn.o): in function `bl602_hbn_irq':
> /Users/runner/work/incubator-nuttx/incubator-nuttx/sources/nuttx/arch/risc-v/src/chip/bl602_hbn.c:91: undefined reference to `bl602_hbn_clear_rtc_int'
> make[1]: *** [nuttx] Error 1
> make: *** [nuttx] Error 2
> make: Target `all' not remade because of errors.
> ```

Addressed, Thank you!

File: Kconfig [hunk 1]
```diff
@@ -1956,6 +1956,19 @@ config DEBUG_OPTLEVEL
 		This string represents the custom optimization level that will be
 		used if DEBUG_CUSTOMOPT.
 
+config DEBUG_OPT_UNUSED_SECTIONS
+	bool "Optimization to eliminate the unused input sections"
+	default y
+	depends on ARCH_TOOLCHAIN_GNU
```
Comment (xiaoxiang781216) [inline Kconfig:1962]: remove ARCH_TOOLCHAIN_GNU, since ez80 support the data/function section but it isn't gcc derived.

**Comment by xiaoxiang781216** [comment]
The change is already contained in https://github.com/apache/incubator-nuttx/pull/6121, let's close this PR.

**Pull request closed without merging**